LEAF_Request_Portal/sources/System.php

```diff
@@ -922,4 +922,396 @@ public function getHistory($filterById)
+
+            $leaderGroupID = $nexus_group->getGroupLeader($group['groupID']);
+            $query_counter++;
+
+            $employees = array_merge($nexus_position->getEmployees($leaderGroupID), $nexus_group->listGroupEmployees($group['groupID']));
+            $query_counter++;
+            $query_counter++;
+
+            foreach ($employees as $employee) {
+                $nexus_users[$group_counter]['userID'] = $employee['userName'];
+                $nexus_users[$group_counter]['groupID'] = $group['groupID'];
+                $nexus_users[$group_counter]['backupID'] = null;
+
+                $group_counter++;
+
+                if (isset($employee['backups'])) {
+                    foreach ($employee['backups'] as $backup) {
+                        $nexus_users[$group_counter]['userID'] = $backup['userName'];
+                        $nexus_users[$group_counter]['groupID'] = $group['groupID'];
+                        $nexus_users[$group_counter]['backupID'] = $employee['userName'];
+                    $group_counter++;
+                    }
+                }
+            }
+
+
+            $counter++;
+        }
+
+        $groups = $this->getOrgchartImportTags($nexus_group);
+        $query_counter++;
+
+        foreach ($groups as $group) {
+            $nexus_groups[$counter]['groupID'] = $group['groupID'];
+            $nexus_groups[$counter]['parentGroupID'] = null;
+            $nexus_groups[$counter]['name'] = $group['groupTitle'];
+
+            $leaderGroupID = $nexus_group->getGroupLeader($group['groupID']);
+            $query_counter++;
+
+            $employees = array_merge($nexus_position->getEmployees($leaderGroupID), $nexus_group->listGroupEmployees($group['groupID']));
+            $query_counter++;
+            $query_counter++;
+
+            foreach ($employees as $employee) {
+                $nexus_users[$group_counter]['userID'] = $employee['userName'];
+                $nexus_users[$group_counter]['groupID'] = $group['groupID'];
+                $nexus_users[$group_counter]['backupID'] = null;
+
+                $group_counter++;
+
+                $backups = $nexus_employee->getBackups($employee['empUID']);
+                $query_counter++;
+                foreach ($backups as $backup) {
+                    $nexus_users[$group_counter]['userID'] = $backup['userName'];
+                    $nexus_users[$group_counter]['groupID'] = $group['groupID'];
+                    $nexus_users[$group_counter]['backupID'] = $employee['userName'];
+                    $group_counter++;
+                }
+            }
+
+            $counter++;
+        }
+
+        $portal_groups = $org_group->getAllGroups();
+        $portal_users = $org_group->getAllUsers();
+
+        $query_counter++;
+        $query_counter++;
+
+        $query_counter += $this->processGroups($portal_groups, $portal_users, $nexus_groups, $nexus_users, $org_group);
+
+        echo 'Total number of queries run ' . $query_counter . '<br />';
```

Comment:
Functions should only do one thing: compute or present.  I know there is a lot of extremely sloppy ones we inherited.  However, we need to avoid falling into that.  It's ok to return a string to display, just don't echo them out here.